### DISCUSSION TITLE
Help with Emitting Events in Tauri WebSocket Plugin

### DISCUSSION THREAD

**Discussion by ab-h-i-n** [tauri-apps/tauri#11779] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/35d0f756-4e0b-4212-9937-671744b959af)

I'm trying to emit events from the frontend to the backend using the Tauri WebSocket plugin. The available functions are send, addEventListener, connect, disconnect, and id, but I'm unsure how to emit messages from the backend to the frontend after a connection is established.

using send function i cant specify the event to emit!

Could someone provide guidance on how to properly emit events or send data from the backend to the frontend?

**Comment by FabianLars** [comment]
addListener is what you need to listen to the events/responses that come from the backend https://tauri.app/plugin/websocket/#usage

What you need to send them back from your backend in the first place depends on your websocket server implementation on your backend. Without more info on that i can't really answer that part.

**Reply by ab-h-i-n** [reply]
When we use socket.io client 
we can emit messages from "client" to backend  eg: socket.emit("eventname" , data);
how to do that with tauri websocket plugin

**Reply by FabianLars** [reply, marked as answer]
those events aren't part of the websocket standard and therefore not (yet?) part of the plugin. This also means that constructing the event payload according to https://github.com/socketio/socket.io-protocol yourself _should_ work.

**Reply by ab-h-i-n** [reply]
Thankyou for your help
